### PR TITLE
Update datatree to v0.0.6 [all tests ci]

### PR DESCRIPTION
**Pull request by lsetiawan**
This PR updates the datatree to v0.0.6 as well as updating the usage within echopype code.

### CODE REVIEW

**Comment by lsetiawan** [comment]
This PR should fix #732 since now we use datatree 0.0.6 here. @imranmaj could you try out this PR and confirm in your machine that things are working as it should? Thanks!

**Comment by lsetiawan** [comment]
@b-reyes Could you look through the changes to make sure that I didn't miss anything regarding the 0.5 to 0.6 conversion code? Thanks. If all looks good, please approve. 😸

**Comment by b-reyes** [comment]
@lsetiawan do you know why this is failing the tests?

**Comment by lsetiawan** [comment]
The test fails are unrelated to this PR. It's crashing workers on plotting tests.. Was gonna troubleshoot that but it's Friday and ran out of time.

File: echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py [hunk 1]
```diff
@@ -105,13 +105,11 @@ def _reorganize_beam_groups(ed_obj):
 
     # Map Beam --> Sonar/Beam_group1
     if "Beam" in ed_obj.group_paths:
-        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam"])
-        ed_obj._tree["Sonar/Beam"].name = "Beam_group1"
+        ed_obj._tree["Sonar/Beam_group1"] = ed_obj._tree["Beam"]
```
Comment (b-reyes) [inline echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py:108]: @lsetiawan could you explain to me why this change needed to occur? It looks like this change is the reason you modify other portions of the conversion code.

**Comment by leewujung** [comment]
> The test fails are unrelated to this PR. It's crashing workers on plotting tests.. Was gonna troubleshoot that but it's Friday and ran out of time.

Hey @lsetiawan : this one: https://github.com/OSOceanAcoustics/echopype/runs/7157410407?check_suite_focus=true#step:15:1148 looks like something different? Although it has the same worker failed situation but it is not in plotting. Could you take a look? Thanks.

File: echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py [hunk 2]
```diff
@@ -985,7 +984,9 @@ def _rename_vendor_group(ed_obj):
     The function directly modifies the input EchoData object.
     """
 
-    ed_obj._tree["Vendor"].name = "Vendor_specific"
+    node = ed_obj._tree["Vendor"]
+    ed_obj._tree["Vendor"].orphan()
```
Comment (b-reyes) [inline echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py:988]: @lsetiawan can you explain to me why this is necessary?

**Comment by b-reyes** [comment]
@lsetiawan and @leewujung after reviewing `test_v05x_v06x_conversion_structure`, I remembered that the tests for the beam groups are quite comprehensive. So, I think we can be confident that @lsetiawan's changes to the conversion routine for the beam groups are safe. The test for the `Vendor_specific` group is less comprehensive because the attributes are somewhat unstable, so we can't directly check them. For this reason, I am unsure if your change is appropriate (please see my inquiry above).  

As long as `test_v05x_v06x_conversion_structure` passes locally, I think we can go ahead and merge this. If you want me to look into this further, I can do it next week.

**Comment by lsetiawan** [comment]
> Although it has the same worker failed situation but it is not in plotting. 

This seems to not happen for Python 3.8 and 3.10. So I'm checking to see if it's intermittent like the plotting ones. 😞

File: echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py (hunk 2 above)
Comment (lsetiawan) [reply echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py:988]: This is needed to remove the old group... seems to work this way now. Maybe can look into how to improve it in future.

**Comment by lsetiawan** [comment]
Looks like it's intermittent. Now the crashing is back to plotting for Python 3.9.

**Comment by leewujung** [comment]
Ok, all tests passed locally. I'll merge this so that we can get to a semi-final version of v0.6.1 to catch potential conflicts in merging to `main`. 

The lingering questions above I think we can get to the bottom of them when working on #708.
